visualization/app/codeCharta/ui/artificialIntelligence/artificialIntelligence.component.ts

```diff
@@ -40,34 +40,37 @@ interface MetricSuggestionParameters {
 	isOutlier?: boolean
 }
 
+interface RiskProfile {
+	lowRisk: number
+	moderateRisk: number
+	highRisk: number
+	veryHighRisk: number
+	totalRloc?: number
+}
+
+const HEIGHT_METRIC = "mcc"
+const AREA_METRIC = "rloc"
+const EXCLUDED_FILES = new Set(["html", "sass", "scss", "scss", ".txt", "md", "json", undefined])
+
 export class ArtificialIntelligenceController
 	implements FilesSelectionSubscriber, BlacklistSubscriber, ExperimentalFeaturesEnabledSubscriber
 {
 	private _viewModel: {
 		analyzedProgrammingLanguage: string
 		suspiciousMetricSuggestionLinks: MetricSuggestionParameters[]
 		unsuspiciousMetrics: string[]
-		riskProfile: {
-			lowRisk: number
-			moderateRisk: number
-			highRisk: number
-			veryHighRisk: number
-		}
+		riskProfile: RiskProfile
 	} = {
 		analyzedProgrammingLanguage: undefined,
 		suspiciousMetricSuggestionLinks: [],
 		unsuspiciousMetrics: [],
-		riskProfile: {
-			lowRisk: 0,
-			moderateRisk: 0,
-			highRisk: 0,
-			veryHighRisk: 0
-		}
+		riskProfile: {} as RiskProfile
 	}
 
-	private debounceCalculation: () => void
+	private rlocRisk = {} as RiskProfile
```

Comment:
I don't understand yet why we have two RiskProfiles. We have one in the viewModel, which represents what we render at the end and we also have this class variable here.
I assume, that this class variable `rlocRisk` is just required for some internal calculations before transferring it to the viewModel. (Correct me if I'm wrong).

If that's the case, we should try to only work with the viewModel here.